server/aspnetcore/TicketHub/Program.cs

```diff
@@ -0,0 +1,55 @@
+using Microsoft.AspNetCore.Diagnostics;
+using Newtonsoft.Json;
+using TicketHub.Database;
+using Styra.Opa;
+using Styra.Opa.AspNetCore;
+using Microsoft.AspNetCore.Authentication;
+
+
+string opaURL = System.Environment.GetEnvironmentVariable("OPA_URL") ?? "http://localhost:8181";
+OpaClient opa = new OpaClient(opaURL);
+
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the builder.
+builder.Services.AddDbContext<PostgresContext>();
+builder.Services.AddLogging(builder => builder.SetMinimumLevel(LogLevel.Trace).AddConsole());
+builder.Services.AddControllers().AddNewtonsoftJson(opts =>
+{
+    opts.SerializerSettings.ReferenceLoopHandling = ReferenceLoopHandling.Ignore;
+    opts.SerializerSettings.NullValueHandling = NullValueHandling.Ignore;
+});
+builder.Services.AddExceptionHandler<CustomExceptionHandler>();
+builder.Services.AddAuthentication(options =>
+{
+    options.DefaultAuthenticateScheme = "CustomCookieAuth";
+    options.DefaultChallengeScheme = "CustomCookieAuth";
+}).AddScheme<AuthenticationSchemeOptions, CustomCookieAuthenticationHandler>(
+    "CustomCookieAuth", options => { });
+
+// Create the top-level application.
+var app = builder.Build();
+
+// Wire in middleware.
+app.UseAuthentication();
+app.UseMiddleware<OpaAuthorizationMiddleware>(opa, "tickets/aspnetcore/main");

```

Comment:
[server change]: Here's where the [ASP.NET Core SDK](https://github.com/StyraInc/opa-aspnetcore) plugs in! We reference a framework-specific policy, so that we can deal with its quirks more easily.